odd-collector-sdk/odd_collector_sdk/collector.py

```diff
@@ -59,8 +65,52 @@ def __init__(
         plugins_package: str = "adapters",
     ) -> None:
         print_collector_packages_info(root_package)
-        self.config = load_config(config_path, plugin_factory)
 
+        # Parse collector_config.yaml
+        parsed_config = read_config_yaml(config_path)
+        (
+            secrets_info,
+            local_collector_settings,
+            local_plugins,
+        ) = unpack_config_logical_sections(parsed_config)
+
+        secrets_backend = secrets_info["secrets_backend"]
+        secrets_backend_kwargs = secrets_info["secrets_backend_kwargs"]
+
+        # Dynamically import the secrets backend class if it is provided
+        if secrets_backend:
+            secrets_backend_module, secrets_backend_class_name = secrets_backend.rsplit(
+                ".", 1
+            )
+            try:
+                secrets_backend_module = import_module(secrets_backend_module)
+                secrets_backend_class = getattr(
+                    secrets_backend_module, secrets_backend_class_name
+                )
+            except ImportError as e:
+                error_message = f"Could not import '{secrets_backend}'"
+                raise ImportError(error_message) from e
+
+            # Retrieve collector config secreats from backend
+            secrets_backend_class_instance = secrets_backend_class(
+                **secrets_backend_kwargs
+            )
+            secret_backend_collector_settings = (
+                secrets_backend_class_instance.get_collector_settings()
+            )
+            secret_backend_plugins = secrets_backend_class_instance.get_plugins()
+        else:
+            secret_backend_collector_settings, secret_backend_plugins = {}, []
```

Comment:
**nit**: We can avoid using `else` here to reduce blocks complexity:
```suggestion
secret_backend_collector_settings, secret_backend_plugins = {}, []

if secrets_backend:
    try:
        secrets_backend_module, secrets_backend_class_name = secrets_backend.rsplit(".", 1)
        secrets_backend_module = import_module(secrets_backend_module)
        secrets_backend_class = getattr(secrets_backend_module, secrets_backend_class_name)
    except ImportError as e:
        raise ImportError(f"Could not import '{secrets_backend}'") from e

    # Retrieve collector config secrets from backend
    secrets_backend_class_instance = secrets_backend_class(**secrets_backend_kwargs)
    secret_backend_collector_settings = secrets_backend_class_instance.get_collector_settings()
    secret_backend_plugins = secrets_backend_class_instance.get_plugins()
```

Reply:
Done